crates/transpiler/src/passes/alap_schedule_analysis.rs

```diff
@@ -0,0 +1,177 @@
+use pyo3::prelude::*;
+use pyo3::types::PyDict;
+use qiskit_circuit::dag_circuit::{DAGCircuit, NodeType, Wire};
+use qiskit_circuit::operations::{Operation, OperationRef};
+use qiskit_circuit::{Clbit, Qubit};
+use rustworkx_core::petgraph::prelude::NodeIndex;
+
+#[pyfunction]
+#[pyo3(name= "alap_schedule_analysis", signature=(dag, clbit_write_latency, node_durations))]
+pub fn run_alap_schedule_analysis(
+    py: Python,
+    dag: &DAGCircuit,
+    clbit_write_latency: u64,
+    node_durations: &Bound<PyDict>,
+) -> PyResult<Py<PyDict>> {
+    if dag.qregs().len() != 1 || !dag.qregs().iter().any(|reg| reg.name() == "q") {
+        return Err(TranspilerError::new_err(
+            "ALAP schedule runs on physical circuits only",
+        ));
+    }
+
+    let mut node_start_time: HashMap<NodeIndex, f64> = HashMap::new();
+    let mut idle_before: HashMap<Wire, f64> = HashMap::new();
+
+    for (i, _) in dag.qubits().objects().iter().enumerate() {
+        idle_before.insert(Wire::Qubit(Qubit::new(i)), 0.0);
+    }
+
+    for (i, _) in dag.clbits().objects().iter().enumerate() {
+        idle_before.insert(Wire::Clbit(Clbit::new(i)), 0.0);
+    }
+
+    for node_index in dag
+        .topological_op_nodes()?
+        .collect::<Vec<_>>()
+        .into_iter()
+        .rev()
+    {
+        let node = dag.get_node(py, node_index)?;
+        let op_duration = match node_durations.get_item(node) {
+            Ok(Some(val)) => val.extract::<f64>()?,
+            Ok(None) => return Err(TranspilerError::new_err("Node duration not found for node")),
+            Err(e) => {
+                return Err(TranspilerError::new_err(format!(
+                    "PyDict get_item error: {}",
+                    e
+                )))
+            }
+        };
+
+        let op = match dag.dag().node_weight(node_index) {
+            Some(NodeType::Operation(op)) => op,
+            _ => continue,
+        };
+
+        let qargs: Vec<Wire> = dag
+            .qargs_interner()
+            .get(op.qubits)
+            .iter()
+            .map(|&q| Wire::Qubit(q))
+            .collect();
+        let cargs: Vec<Wire> = dag
+            .cargs_interner()
+            .get(op.clbits)
+            .iter()
+            .map(|&c| Wire::Clbit(c))
+            .collect();
+
+        /*
+         compute t0, t1: instruction interval, note that
+         t0: start time of instruction
+         t1: end time of instruction
+
+         since this is alap scheduling, node is scheduled in reversed topological ordering
+         and nodes are packed from the very end of the circuit.
+         the physical meaning of t0 and t1 is flipped here.
+        */
+
+        // Get operation type
+        let op_name = op.op.name();
+        let op_view = op.op.view();
+        let (is_gate, is_delay) = match op_view {
+            OperationRef::Gate(_) => (true, false),
+            OperationRef::StandardGate(_) => (true, false),
+            _ => (false, op_name == "delay"),
```

Comment:
```suggestion
            OperationRef::Gate(_) => (true, false),
            OperationRef::StandardGate(_) => (true, false),
            OperationRef::StandardInstruction(StandardInstruction::Delay(_)) => (false, true),
            _ => (false, op_name == "delay"),
```
You might want to add this condition for completion's sake.